### PR TITLE
FortiManager SSL SSH Security Profiles

### PR DESCRIPTION
**Pull request by ftntcorecse**
##### SUMMARY

Managing SSL and SSH security profiles in FortiManager

##### ISSUE TYPE

- New Module Pull Request

##### COMPONENT NAME

fmgr_secprof_ssl_ssh.py

##### ANSIBLE VERSION

```paste below
ansible 2.6.3
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/home/cse/PycharmProjects/JPMC_FOS/ansible/lib/ansible/modules']
  ansible python module location = /usr/local/lib/python2.7/dist-packages/ansible
  executable location = /usr/local/bin/ansible
  python version = 2.7.12 (default, Dec  4 2017, 14:50:18) [GCC 5.4.0 20160609]
``

##### ADDITIONAL INFORMATION
<!--- Include additional information to help people understand the change here -->
<!--- A step-by-step reproduction of the problem is helpful if there is no related issue -->

<!--- Paste verbatim command output below, e.g. before and after your change -->
```paste below

```

### CODE REVIEW

**Comment by ftntcorecse** [comment]
Hi All,

May I have a status update for this PR?  Is anything else needed from me to move this forward?

Thanks

File: lib/ansible/modules/network/fortimanager/fmgr_secprof_ssl_ssh.py [hunk 1]
```diff
@@ -0,0 +1,1095 @@
...
+    description:
+      - The username associated with the account.
+    required: true
+
+  password:
+    description:
+      - The password associated with the username account.
+    required: true
+
+  mode:
```
Comment (justjais) [inline lib/ansible/modules/network/fortimanager/fmgr_secprof_ssl_ssh.py:60]: is this `mode` similar to `state`, if yes can you modify the name to `state` to be in consistent with Ansible modules, 
```suggestion
  state: ['add', 'set', 'delete', 'update'] 
```
for your reference:
```
- name: EXECUTE SCRIPT
  fmgr_script:
    host: "{{inventory_hostname}}"
    username: "{{ username }}"
    password: "{{ password }}"
    adom: "root"
    script_name: "TestScript"
    state: "execute"
    script_scope: "FGT1,FGT2"
```

File: lib/ansible/modules/network/fortimanager/fmgr_secprof_ssl_ssh.py [hunk 2]
```diff
@@ -0,0 +1,1095 @@
...
+      - When enabled, allows SSL sessions whose server certificate validation failed.
+      - choice | disable | Disable setting.
+      - choice | enable | Enable setting.
+    required: false
+    choices: ["disable", "enable"]
+
+  ftps_client_cert_request:
+    description:
+      - Action based on client certificate request failure.
+      - choice | bypass | Bypass.
```
Comment (justjais) [inline lib/ansible/modules/network/fortimanager/fmgr_secprof_ssl_ssh.py:171]:
```suggestion
      - choice : bypass | Bypass.
```

Comment (justjais) [reply lib/ansible/modules/network/fortimanager/fmgr_secprof_ssl_ssh.py:171]: For all the choice.

File: lib/ansible/modules/network/fortimanager/fmgr_secprof_ssl_ssh.py [hunk 3]
```diff
@@ -0,0 +1,1095 @@
...
+      - Allow, ignore, or block the untrusted SSL session server certificate.
+      - choice | allow | Allow the untrusted server certificate.
+      - choice | block | Block the connection when an untrusted server certificate is detected.
+      - choice | ignore | Always take the server certificate as trusted.
+    required: false
+    choices: ["allow", "block", "ignore"]
+
+  ssl_exempt:
+    description:
+      - EXPERTS ONLY! KNOWLEDGE OF FMGR JSON API IS REQUIRED!
```
Comment (justjais) [inline lib/ansible/modules/network/fortimanager/fmgr_secprof_ssl_ssh.py:566]: To be in consistent to Ansible modules, can you please change the description/format in single statements with formatting like: 
```suggestion
      - EXPERTS ONLY! KNOWLEDGE OF FMGR JSON API IS REQUIRED!
        List of multiple child objects to be added. Expects a list of dictionaries. 
        Dictionaries must use FortiManager API parameters, not the ansible ones listed below. 
        If submitted, all other prefixed sub-parameters ARE IGNORED.
```

File: lib/ansible/modules/network/fortimanager/fmgr_secprof_ssl_ssh.py [hunk 4]
```diff
@@ -0,0 +1,1095 @@
...
+      password: "{{ password }}"
+      name: "Ansible_SSL_SSH_Profile"
+      mode: "delete"
+
+  - name: CREATE Profile
+    fmgr_secprof_ssl_ssh:
+      host: "{{inventory_hostname}}"
+      username: "{{ username }}"
+      password: "{{ password }}"
+      name: "Ansible_SSL_SSH_Profile"
```
Comment (justjais) [inline lib/ansible/modules/network/fortimanager/fmgr_secprof_ssl_ssh.py]: If you are not using variables for particular values, you need not to enclose the values in double quotes, as:
```suggestion
      name: Ansible_SSL_SSH_Profile
```
but, as mentioned above following will remain as it is as its calling variable:
```
host: "{{inventory_hostname}}"
```

File: lib/ansible/modules/network/fortimanager/fmgr_secprof_ssl_ssh.py (hunk 1 above)
Comment (ftntcorecse) [reply lib/ansible/modules/network/fortimanager/fmgr_secprof_ssl_ssh.py:60]: Hi @justjais For our first modules we tried to use the Ansible 'state' parameter, but we stopped using it as it didn't fit with our API model.  We now use mode for which HTTP action to use for our API call.  Thanks

**Comment by ftntcorecse** [comment]
Hi @justjais I've added fixes to most of your comments.  We are looking at how we can code our future modules using your changes.  Thanks!

File: lib/ansible/modules/network/fortimanager/fmgr_secprof_ssl_ssh.py (hunk 1 above)
Comment (justjais) [reply lib/ansible/modules/network/fortimanager/fmgr_secprof_ssl_ssh.py:60]: @ftntcorecse ok thanks

File: lib/ansible/modules/network/fortimanager/fmgr_secprof_ssl_ssh.py [hunk 5]
```diff
@@ -28,9 +28,9 @@
 module: fmgr_secprof_ssl_ssh
 version_added: "2.8"
 author:
-    - Luke Weighall (@lweighall)
-    - Andrew Welsh (@Ghilli3)
-    - Jim Huber (@p4r4n0y1ng)
+    - Luke Weighall (lweighall)
```
Comment (justjais) [inline lib/ansible/modules/network/fortimanager/fmgr_secprof_ssl_ssh.py]: @ftntcorecse was there any specific reason for removing the `@` as ansible modules follow author with `@` sign.

**Review by justjais — APPROVED** [review]

**Pull request merged**